Crash in SSL

I was playing with ZNC certificates and managed to crash KVIrc 6710797e by connecting it to it while znc.pem is an empty file

The crash happened when I clicked the "connect" button on toolbar in the context which was previously connected to that ZNC instance; at the moment of that click it wasn't connected.

Here's log from ZNC:

```
[2025-03-16 19:51:43.736316] /home/somebody/znc/git/build-cmake/src/Csocket.cpp:1842 Error with SSLCert file [/home/somebody/.znc/znc.pem]
[2025-03-16 19:51:43.736358] /home/somebody/znc/git/build-cmake/src/Csocket.cpp:666 at /home/somebody/znc/git/build-cmake/src/Csocket.cpp:1843
[2025-03-16 19:51:43.736396] /home/somebody/znc/git/build-cmake/src/Csocket.cpp:671 error:0480006C:PEM routines::no start line
[2025-03-16 19:51:43.736425] /home/somebody/znc/git/build-cmake/src/Csocket.cpp:666 at /home/somebody/znc/git/build-cmake/src/Csocket.cpp:1843
[2025-03-16 19:51:43.736457] /home/somebody/znc/git/build-cmake/src/Csocket.cpp:671 error:0A080009:SSL routines::PEM lib
```

I suppose ZNC closed the socket by seeing incorrect certificate, but KVIrc shouldn't crash by that

--- a/src/kvilib/net/KviSSL.cpp
+++ b/src/kvilib/net/KviSSL.cpp
@@ -536,7 +536,15 @@ KviSSL::Result KviSSL::connect()
 {
 	if(!m_pSSL)
 		return NotInitialized;
+#if !(defined(COMPILE_ON_WINDOWS) || defined(COMPILE_ON_MINGW))
+		// ignore SIGPIPE
+		signal(SIGPIPE, SIG_IGN);
+		int ret = SSL_connect(m_pSSL);
+		//restore normal SIGPIPE behaviour.
+		signal(SIGPIPE, SIG_DFL);
+#else
 	int ret = SSL_connect(m_pSSL);
+#endif
 	return connectOrAcceptError(ret);
 }
 

